src/particles/elements/RFCavity.H

```diff
@@ -0,0 +1,425 @@
+
+              refpart.map(3,3) = (1.0_prt-s*L)*R(3,3) + s*R(4,3);
+              refpart.map(3,4) = (1.0_prt-s*L)*R(3,4) + s*R(4,4);
+              refpart.map(4,3) = -s*pow(L,2)*R(3,3) + (1.0_prt+s*L)*R(4,3);
+              refpart.map(4,4) = -s*pow(L,2)*R(3,4) + (1.0_prt+s*L)*R(4,4);
+
+        }
+
+        void map1 (amrex::ParticleReal const tau,
+                   RefPart & refpart,
+                   amrex::ParticleReal & zeval) const
+        {
+
+            using namespace amrex::literals; // for _rt and _prt
+
+
+               amrex::ParticleReal const t = refpart.t;
+               amrex::ParticleReal const pt = refpart.pt;
+               amrex::ParticleReal const z = zeval;
+
+               // Define parameters and intermediate constants
+               amrex::ParticleReal const pi = 3.141592653589793_prt;
+               amrex::ParticleReal const c = 2.99792458e8;
+               amrex::ParticleReal const k = 2.0_prt*pi*m_freq/c;
+               amrex::ParticleReal const phi = m_phase*pi/180.0_prt;
+               amrex::ParticleReal const E0 = m_Escale;
+
+               // push the reference particle
+
+               auto[ez,ezp,ezint] = RF_Efield(z);
+               zeval = z + tau;
+               auto[ezf,ezpf,ezintf] = RF_Efield(zeval);
+
+               refpart.t = t;
+               refpart.pt = pt - E0*(ezintf-ezint)*cos(k*t+phi);
+               amrex::AllPrintToFile("rfout") << z << " " << ez << "\n";
+
+               // push the linear map equations
+
+              amrex::Array2D<amrex::ParticleReal,1,6,1,6> const R = refpart.map;
+              amrex::ParticleReal const M = E0*(ezintf-ezint)*k*sin(k*t+phi);
+              amrex::ParticleReal const L = E0*(ezpf-ezp)*sin(k*t+phi)/(2.0_prt*k)+M/2.0_prt;
+
+              refpart.map(2,1) = L*R(1,1) + R(2,1);
+              refpart.map(2,2) = L*R(1,2) + R(2,2);
+
+              refpart.map(4,3) = L*R(3,3) + R(4,3);
+              refpart.map(4,4) = L*R(3,4) + R(4,4);
+
+              refpart.map(6,5) = M*R(5,5) + R(6,5);
+              refpart.map(6,6) = M*R(5,6) + R(6,6);
+
+        }
+
+        /** Number of slices used for the application of space charge
+         *
+         * @return positive integer
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        int nslice () const
+        {
+            return m_nslice;
+        }
+
+        /** Return the segment length
+         *
+         * @return value in meters
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        amrex::ParticleReal ds () const
+        {
+            return m_ds;
+        }
+
+    private:
+        amrex::ParticleReal m_ds; //! segment length in m
+        amrex::ParticleReal m_Escale; //! scaling factor for RF electric field
+        amrex::ParticleReal m_freq; //! RF frequency in Hz
+        amrex::ParticleReal m_phase; //! RF driven phase in deg
+        amrex::ParticleReal m_Bzsol; //! strength of solenoid field in T
```

Comment:
Is this missing in the implementation? Currently, all external solenoid fields are ignored:
```
private field 'm_Bzsol' is not used [-Werror,-Wunused-private-field]
```